Fix backup restore

TeddyAndrieux opened the pull request:
**Component**:

'backup', 'restore'
<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

- Backup archive layout was wrong
- Restore not longer working since #2103 

**Summary**:

- Add abilty to provide apiserver ip from pillar to configure apiserver proxy and admin kubeconfig
- Change backup archive layout
- Add some metadata + integrity check for backup archive
- Add `apiserver_ip` arg in restore script 

**Acceptance criteria**: 

Working backup restore procedure

TODO:

- [x] Update backup restore documentation

---

Fixes: #2141 
Fixes: #2142 
Fixes: #2143 
Fixes: #2157

TeddyAndrieux committed:
Revert "Certificates: Add backup and restore for Metalk8s CA's"

This reverts commit 3f610a5f2491160dcd92315fa58ed6e3c42bf255.
This change are not needed as we already backup all `/etc/metalk8s`
directory that contains all the "MetalK8s CAs" in `/etc/metalk8s/pki`

Fixes: #2141

TeddyAndrieux committed:
scripts: Reflect machine layout in backup archive

Move kubernetes pki from `./pki` to `./kubernetes/pki` in backup archive
to reflect machine layout

Fixes: #2142

TeddyAndrieux committed:
scripts: Add some `metadata` and `sha256sum` in backup archive

Add some metadata information in the backup archive and a `sha256sum` of
all the file, and check integrity of the backup archive at the beginning
of the restore script

Fixes: #2143

TeddyAndrieux committed:
scripts: Add `apiserver_ip` as argument for restore script

Since #2103 we not longer use VIP for apiserver so we need an IP of one
apiserver to be able to register the new bootstrap node to the current
kubernetes cluster

Fixes: #2157

TeddyAndrieux committed:
salt: Add ability to give `apiserver_ip` from pillar

For restore we may want to specify an external apiserver IP for
apiserver proxy and admin kubeconfig

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/bugfix/fix-backup-restore` with contents from `bugfix/fix-backup-restore`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/bugfix/fix-backup-restore origin/development/2.5
 $ git merge origin/bugfix/fix-backup-restore
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/bugfix/fix-backup-restore
```

TeddyAndrieux commented:
/help

bert-e commented:
# Help page



The following options and commands are available at this time.

## Options ##


name   | description  | privileged   | authored
------ | ------------ | ------------ |----------
:arrow_right: **create_pull_requests** | Allow the creation of integration pull requests. |  | 
:arrow_right: **bypass_author_approval** | Bypass the pull request author's approval |  :star:  | 
:arrow_right: **bypass_commit_size** | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
:arrow_right: **unanimity** | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
:arrow_right: **bypass_incompatible_branch** | Bypass the check on the source branch prefix |  :star:  | 
:arrow_right: **bypass_build_status** | Bypass the build and test status |  :star:  | 
:arrow_right: **approve** | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
:arrow_right: **after_pull_request** | Wait for the given pull request id to be merged before continuing with the current one. |  | 
:arrow_right: **wait** | Instruct Bert-E not to run until further notice. |  | 
:arrow_right: **bypass_jira_check** | Bypass the Jira issue check |  :star:  | 
:arrow_right: **no_octopus** | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
:arrow_right: **bypass_leader_approval** | Bypass the pull request leaders' approval |  :star:  | 
:arrow_right: **bypass_peer_approval** | Bypass the pull request peers' approval |  :star:  | 



## Commands ##


name   | description  | privileged
------ | ------------ | ------------
:arrow_right: **reset** | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
:arrow_right: **help** | Print Bert-E's manual in the pull request. | 
:arrow_right: **build** | Re-start a fresh build ```TBA``` | 
:arrow_right: **force_reset** | Delete integration branches & pull requests, and restart merge process from the beginning. | 
:arrow_right: **retry** | Re-start a fresh build ```TBA``` | 
:arrow_right: **status** | Print Bert-E's current status in the pull request ```TBA``` | 
:arrow_right: **clear** | Remove all comments from Bert-E from the history ```TBA``` |

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/bugfix/fix-backup-restore` with contents from `bugfix/fix-backup-restore`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/bugfix/fix-backup-restore origin/development/2.5
 $ git merge origin/bugfix/fix-backup-restore
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/bugfix/fix-backup-restore
```

TeddyAndrieux commented:
∕after_pull_request 2147

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

TeddyAndrieux commented:
/after_pull_request=2147

bert-e commented:
# Waiting for other pull request(s)



The current pull request is locked by the **after_pull_request** option.

In order for me to merge this pull request, run the following actions first:


:arrow_right: Merge the `OPEN` pull request:

* pull request #2147





Alternatively, delete all the **after_pull_request** comments from this pull request.





*The following options are set:* **after_pull_request**

TeddyAndrieux commented:
/approve

bert-e commented:
# Waiting for other pull request(s)



The current pull request is locked by the **after_pull_request** option.

In order for me to merge this pull request, run the following actions first:


:arrow_right: Merge the `OPEN` pull request:

* pull request #2147





Alternatively, delete all the **after_pull_request** comments from this pull request.





*The following options are set:* **approve, after_pull_request**

gdemonet commented on scripts/restore.sh.in:
```diff
@@ -97,7 +102,7 @@ extract_archive() {
 
 check_integrity() {
-    ( cd "$BACKUP_DIR" && sha256sum -c .metadata )
+    ( cd "$BACKUP_DIR" && sha256sum -c .sha256sum )
```
This should be in previous commit :/

gdemonet commented on docs/operation/bootstrap_backup_restore.rst:
```diff
@@ -41,9 +35,9 @@ First mount the ISO and then run the restore script:
 
     Replace `<backup_archive>` with the path to the backup archive you want
-    to use.
+    to use and `<node_ip>` with a node IP of one apiserver.
```
Maybe mention "control-plane IP" and refer to simpler concepts (control-plane Node instead of `apiserver`, or at least refer to the Glossary).

gdemonet commented on salt/metalk8s/kubernetes/apiserver-proxy/files/apiserver-proxy.conf.j2:
```diff
@@ -10,12 +10,16 @@ events {
-        fun='control_plane_ip',
-        tgt_type='compound',
-    )
+{%- if pillar.get('apiserver_ip') %}
+{%-     set masters_request = {'ip': pillar.apiserver_ip} %}
```
Why not squash with the previous commit?

TeddyAndrieux replied:
What do you mean ?
Previous commit is about adding metadata in backup archive it's not related to this

gdemonet replied:
In GitHub history, we get:

```
- scripts: Add `apiserver_ip` as argument for restore script
- salt: Add ability to give `apiserver_ip` from pillar (this commit)
```

My question was: why not keep them together? I don't see another reason to use this pillar value other than during the restore script, so keeping this as a single commit would help readers understand why it was introduced.

TeddyAndrieux replied:
IMO make sense to split them as we add the feature to being able to specify a custom "apiserver IP" from pillar and then we use this feature in the restore script maybe I should just change the commit message of the first one

gdemonet reviewed (commented):
Most of it looks good to me, will test :)

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **approve, after_pull_request**

alexandre-allard commented:
I don't know if it's a PEBKAC issue (or even a flaky), but I had to add the new etcd member by hand to complete the bootstrap node restoration.
What did I do (on a 2.4.2-dev, 2 nodes + bootstrap cluster) ?
- Deleted the boostrap host
- Removed it from etcd members
```
crictl exec -i "f4ea35c86078e" sh -c "ETCDCTL_API=3 etcdctl --endpoints https://127.0.0.1:2379 --cert /etc/kubernetes/pki/etcd/server.crt --key /etc/kubernetes/pki/etcd/server.key --cacert /etc/kubernetes/pki/etcd/ca.crt member remove 77170215bd1dd1d4"
```
- Spawned a new bootstrap host
- Launched restore.sh script (script failed on `metalk8s.orchestrate.deploy_node`, not able to get an answer from the API server):
```
/var/tmp/metalk8s/restore.sh -b /tmp/backup_20200106_105508.tar.gz -i 192.168.1.5 -v
```
  Here is the output of the script:
```
[...]
[ERROR   ] {u'return': {u'outputter': u'highstate', u'data': {u'metalk8s-tznqr-ootstrap.novalocal_master': [u'Pillar failed to render with the following messages:', u"Failed to load ext_pillar metalk8s_nodes: HTTPSConnectionPool(host='192.168.1.11', port=6443): Max retries exceeded with url: /api/v1/nodes (Caused by ProtocolError('Connection aborted.', error(111, 'Connection refused')))"]}, u'retcode': 1}}
metalk8s-tznqr-bootstrap.novalocal_master:
  Name: Set grains - Function: salt.state - Result: Clean Started: - 09:30:10.137568 Duration: 4374.543 ms
  Name: mine.update - Function: salt.function - Result: Changed Started: - 09:30:14.512459 Duration: 768.186 ms
  Name: metalk8s-tznqr-bootstrap.novalocal - Function: metalk8s_cordon.node_cordoned - Result: Changed Started: - 09:30:15.282085 Duration: 90.562 ms
  Name: metalk8s-tznqr-bootstrap.novalocal - Function: metalk8s_drain.node_drained - Result: Changed Started: - 09:30:15.374478 Duration: 4431.238 ms
  Name: saltutil.sync_all - Function: salt.function - Result: Changed Started: - 09:30:19.805975 Duration: 927.459 ms
  Name: metalk8s.check_pillar_keys - Function: salt.function - Result: Changed Started: - 09:30:20.734152 Duration: 30772.793 ms
  Name: Reconfigure salt-minion - Function: salt.state - Result: Changed Started: - 09:30:51.545540 Duration: 12682.587 ms
  Name: metalk8s_saltutil.wait_minions - Function: salt.runner - Result: Changed Started: - 09:31:04.228859 Duration: 2894.404 ms
  Name: Run the highstate - Function: salt.state - Result: Changed Started: - 09:31:07.124583 Duration: 42952.273 ms
----------
          ID: Wait for API server to be available
    Function: http.wait_for_successful_query
        Name: https://127.0.0.1:7443/healthz
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1919, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1918, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/http.py", line 163, in wait_for_successful_query
                  raise caught_exception  # pylint: disable=E0702
              SSLEOFError: EOF occurred in violation of protocol (_ssl.c:618)
     Started: 09:31:50.079001
    Duration: 300002.151 ms
     Changes:
----------
          ID: Uncordon the node
    Function: metalk8s_cordon.node_uncordoned
        Name: metalk8s-tznqr-bootstrap.novalocal
      Result: False
     Comment: One or more requisite failed: metalk8s.orchestrate.deploy_node.Wait for API server to be available
     Started: 09:36:50.082547
    Duration: 0.019 ms
     Changes:
  Name: ps.pkill - Function: salt.function - Result: Changed Started: - 09:36:50.082811 Duration: 376.399 ms
----------
          ID: Register the node into etcd cluster
    Function: salt.runner
        Name: state.orchestrate
      Result: False
     Comment: Runner function 'state.orchestrate' failed.
     Started: 09:36:50.459635
    Duration: 9033.168 ms
     Changes:
              ----------
              return:
                  ----------
                  data:
                      ----------
                      metalk8s-tznqr-bootstrap.novalocal_master:
                          - Pillar failed to render with the following messages:
                          - Failed to load ext_pillar metalk8s_nodes: HTTPSConnectionPool(host='192.168.1.11', port=6443): Max retries exceeded with url: /api/v1/nodes (Caused by ProtocolError('Connection aborted.', error(111, 'Connection refused')))
                  outputter:
                      highstate
                  retcode:
                      1

Summary for metalk8s-tznqr-bootstrap.novalocal_master
-------------
Succeeded: 10 (changed=8)
Failed:     3
-------------
Total states run:     13
Total run time:  409.306 s
```
  In etcd container logs, I have:
```
etcdmain: error validating peerURLs {ClusterID:e8b3e49ae5233ad1 Members:[&{ID:df96615de5709ef8 RaftAttributes:{PeerURLs:[https://192.168.1.32:2380]} Attributes:{Name:metalk8s-tznqr-node-2.novalocal ClientURLs:[https://192.168.1.32:2379]}} &{ID:e9f15dbabe9226d5 RaftAttributes:{PeerURLs:[https://192.168.1.5:2380]} Attributes:{Name:metalk8s-tznqr-node-1.novalocal ClientURLs:[https://192.168.1.5:2379]}}] RemovedMemberIDs:[]}: member count is unequal
```
- Tried to relaunch, still failing, even earlier.
- Added the etcd member by hand (on a healthy node)
```
crictl exec -i "f4ea35c86078e" sh -c "ETCDCTL_API=3 etcdctl --endpoints https://127.0.0.1:2379 --cert /etc/kubernetes/pki/etcd/server.crt --key /etc/kubernetes/pki/etcd/server.key --cacert /etc/kubernetes/pki/etcd/ca.crt member add metalk8s-tznqr-bootstrap.novalocal --peer-urls=https://192.168.1.11:2380"
```
- Relaunched the restore script (everything went fine this time and the node is correctly restored).

Am I missing a step or something ?
Maybe it's not related to this PR, but an issue during etcd expansion or something like that.

gdemonet commented on docs/operation/bootstrap_backup_restore.rst:
```diff
@@ -20,12 +20,6 @@ Backup archives are stored in `/var/lib/metalk8s/`.
-    You cannot use the restore script if you do not have High Availability
-    apiserver because some information required to reconfigure the others
-    nodes are stored in the apiserver.
-
```
Thinking that we should still warn that a restoration procedure needs a working K8s API. So maybe we don't need the actual HA setup for `k-a`, but we need a highly available control plane, with a 3 nodes `etcd` cluster at a minimum to make this restoration procedure work. *Or* have the single-node control plane distinct from the Bootstrap Node.

TeddyAndrieux replied:
Make sense I will add it

gdemonet commented on docs/operation/bootstrap_backup_restore.rst:
```diff
@@ -20,12 +20,6 @@ Backup archives are stored in `/var/lib/metalk8s/`.
-
 .. warning::
 
     In case of a 3-node etcd cluster (2 nodes + unreachable old bootstrap node)
```
Good point, maybe it should be an optional step in the clear, with the required command(s) and expected output. Too easy to go over it and have weird errors in the script's output.

TeddyAndrieux replied:
Done but command are quite ugly we could do something smarter if we add a salt module to remove an etcd member but not part of this PR IMO

gdemonet replied:
Indeed, would be nice to provide a user-friendly wrapper to `etcdctl`, maybe as a Salt module or in a `kubectl` plugin, not sure. For now, having the "ugly" command for copy-pasting is good enough!

bert-e commented:
# History mismatch



Merge commit #bc1e12933d05b44283ebcf6fff18face66d09be6 on the integration branch
`w/2.5/bugfix/fix-backup-restore` is merging a branch which is neither the current
branch `bugfix/fix-backup-restore` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `bugfix/fix-backup-restore` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve, after_pull_request**

TeddyAndrieux committed:
scripts: Add `mine.update` after minion configuration in restore

To properly configure the new bootstrap node we need to have mine
informations from other nodes (like control plane and workload plane
IPs), to be able to update the mine we need that all minions ready and
having correct pillar

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve, after_pull_request**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/bugfix/fix-backup-restore` with contents from `bugfix/fix-backup-restore`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/bugfix/fix-backup-restore origin/development/2.5
 $ git merge origin/bugfix/fix-backup-restore
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/bugfix/fix-backup-restore
```





*The following options are set:* **approve, after_pull_request**

TeddyAndrieux committed:
scripts: Restore, add a `sync_auth` and reconfigure CP ingress

During restore we need to sync_auth to have a working salt-api, also
need to reconfigure the control plane nginx ingress to set the new
external IP used to access the UI

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort










*The following options are set:* **approve, after_pull_request**

gdemonet commented on docs/operation/bootstrap_backup_restore.rst:
```diff
@@ -23,14 +23,34 @@ Restoration procedure
     You cannot use the restore script if you do not have High Availability
-    apiserver because some information required to reconfigure the others
-    nodes are stored in the apiserver.
+    control plane, with at least a 3-node etcd cluster.
```
Let's try wording it without "You":

```
It is mandatory to have a highly available control plane, with at least
3 members in the ``etcd`` cluster (including the failed bootstrap Node),
to use the restore script.
```

gdemonet commented on docs/operation/bootstrap_backup_restore.rst:
```diff
@@ -23,14 +23,34 @@ Restoration procedure
 
-.. warning::
+In case of a 3-node etcd cluster (2 nodes + unreachable old bootstrap node)
+you need to remove the old bootstrap node from the etcd cluster before
```
We just said it's mandatory, so let's get rid of the "In case of", and reword a bit:

```
Before running the script, the unreachable ``etcd`` member needs to be
unregistered from the cluster. To do so, run the following commands
from a working Node with the ``etcd`` role:

.. code::

   [...]
```

TeddyAndrieux replied:
This step is mandatory only for a 3 node etcd if you have 4 etcd node you don't need to remove the old one (btw you should remove it I think but it's not needed)

So yes I will change it to always remove it

gdemonet reviewed (changes requested):
Just some docs changes, otherwise LGTM

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet







*The following options are set:* **approve, after_pull_request**

bert-e commented:
# History mismatch



Merge commit #09939e4df6c156c48413af0f463e514a8dac89ee on the integration branch
`w/2.5/bugfix/fix-backup-restore` is merging a branch which is neither the current
branch `bugfix/fix-backup-restore` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `bugfix/fix-backup-restore` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve, after_pull_request**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve, after_pull_request**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.5/bugfix/fix-backup-restore` with contents from `bugfix/fix-backup-restore`
and `development/2.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.5/bugfix/fix-backup-restore origin/development/2.5
 $ git merge origin/bugfix/fix-backup-restore
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/bugfix/fix-backup-restore
```





*The following options are set:* **approve, after_pull_request**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet







*The following options are set:* **approve, after_pull_request**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/2e05ad8fe78f0a9b5fe090b9963e9218da6dbb10) did not succeed in branch w/2.5/bugfix/fix-backup-restore.




*The following options are set:* **approve, after_pull_request**

gdemonet commented on scripts/restore.sh.in:
```diff
@@ -270,13 +291,22 @@ highstate_bootstrap() {
+    SALT_MASTER_CALL=(crictl exec -i "$(get_salt_container)")
+
+    "${SALT_MASTER_CALL[@]}" salt-run --state-output=mixed state.orchestrate \
+        metalk8s.addons.nginx-ingress-control-plane.deployed \
```
Maybe we could simply run `metalk8s.deployed` here? Being idempotent and only executed on the master, I'm thinking it's some kind of "K8s highstate"... If you think so, we could do with a debt ticket to fix in 2.5 :)

TeddyAndrieux replied:
Since for the moment we basically replace every object I think it's not a good idea because we will lose all tuning done by the user on k8s objects when restoring which is unexpected IMO

gdemonet replied:
Good point, let's wait for #1996 then

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve, after_pull_request**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye teddyandrieux.
